crates/aptos-dkg/tests/barycentric_eval.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_dkg::algebra::polynomials::barycentric_eval;
+use ark_ff::{FftField, Field};
+use ark_poly::{univariate::DensePolynomial, DenseUVPolynomial, Polynomial};
+use ark_std::UniformRand;
+
+#[test]
+fn test_barycentric_eval() {
+    type F = ark_bls12_381::Fr;
+    let mut rng = ark_std::rand::thread_rng();
+
+    let degree = 64;
+    let poly_coeffs: Vec<F> = (0..=degree).map(|_| F::rand(&mut rng)).collect();
+    let poly = DensePolynomial::from_coefficients_vec(poly_coeffs);
+
+    let n = 128;
+    let omega = F::get_root_of_unity(n).unwrap();
+    let roots: Vec<F> = (0..n).map(|i| omega.pow([i])).collect();
+    let evals: Vec<F> = roots.iter().map(|&root| poly.evaluate(&root)).collect();
+    let n_inv = F::from(n).inverse().unwrap();
+
+    for _ in 0..20 {
```

Comment:
I would lower this to like 5 and iterate through more possibilities for the degree?

Test degree 0 polynomials, including the zero polynomial, degree 1, 2 and, idk, 16.